modules/ossm-automatic-gateway-injection.adoc

```diff
@@ -0,0 +1,51 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v2x/ossm-traffic-manage.adoc
+
+:_content-type: CONCEPT
+[id="ossm-automatic-gateway-injection_{context}"]
+= Enabling gateway injection
+
+Gateway configurations are applied to standalone Envoy proxies that are running at the edge of the mesh, rather than sidecar Envoy proxies running alongside your service workloads. Because gateways are Envoy proxies, you can configure {SMProductShortName} to automatically inject gateways similar to how you can automatically inject sidecars.
+
+Using automatic injection for gateways lets you deploy and manage gateways independent from the `ServiceMeshControlPlane` resource. This lets you manage your gateways with your user applications. Using auto-injection for gateway deployments gives developers full control over the gateway deployment, while also simplifying operations. When a new upgrade is available, or a configuration has changed, you restart the gateway pods to update them. This makes the experience of operating a gateway deployment the same as operating sidecars.
+
+[NOTE]
+====
+Injection is disabled by default for the `ServiceMeshControlPlane` namespace, for example `istio-system`. As a security best practice, deploy gateways in a different namespace from the control plane.
+====
+
+.Prerequisites
+
+* The namespace must have the `istio-injection=enabled` label.
+* The namespace must have the injection webhook enabled.
+
+When deploying a gateway, you must opt-in to injection by adding an injection label or annotation to the gateway `deployment` object.
+
+.Example deployment with annotations
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
We should also list the `Service`, `Role` and `RoleBinding` here, as they're required